db/seeds/development.rb

```diff
@@ -96,3 +96,14 @@
                                                   access_token_audience: 'http://localhost:4000',
                                                   access_token_duration: vaid_access_token_duration,
                                                   certificates: [vaid_certificate])
+
+# Create Service Account Config for Chatbot
+chatbot = SignIn::ServiceAccountConfig.find_or_initialize_by(service_account_id: 'chatbot')
```

Comment:
🗣️ John and I talked this over and we'd like to use service account id's similar to how the client configs use human-readable id's. This will require some additional work to add uniqueness validations and change the Identity Dashboard ID, but isn't a blocker for this PR. Using a human-readable service account ID allows us to easily map the service account to specific actions, such as passing the application name to the MAP STS service from the service account config.

Reply:
I'm fine with human readable ids for localhost (for example, in this case totally fine), but I think we shouldn't have them on other stacks, and some day should transition away from human readable ids for client configs as well. The reason is that these should not be associated with any code logic, and human readable IDs can potentially encourage behavior like 'set application symbol for security token service equal to service_account_id' which is not what we want to encourage.

The 'description' was intended to be the way to encourage a human readable component here

Reply:
Hmmm, the issue I'm running into is that the service requires a symbolized name that it then maps to a configured ID for an application [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/map/security_token/service.rb#L43). We do have a few options for making this work:

1) We could refactor the `token_params` method to check if the provided value is a configured client ID or, if the value does not correspond to a configured client ID, attempt to retrieve the client ID from the application name mapping as a fallback.

2) We could add an additional keyword argument to the `token` method (maybe `client_id`) that defaults to nil, but is used to skip the application config mapping if provided.

3) We could refactor the `token` method to accept `client_id` instead of `application`, pushing the burden of determining the correct client id to the calling component.

I think my preference is 3, but the other options seem valid and I'm sure there are other approaches we could take, too. What are your thoughts?

Reply:
So the problem with option 3 here is that the `application` could potentially come from an external service (which would have a `service_account_id`, or it could come from just any random component inside `vets-api`, which wouldn't have a `service_account_id`. `vets-api` could theoretically use the same `application` for all internal calls, but we decided we wanted more insight into who was using this service than that

Reply:
Oh how about this:

Can we encode the application name in a path param and then make it the scope?

`map_services/:application/token`, and then the scope in the service account config would be: `http://localhost:3000/v0/map_services/chatbot/token`

Reply:
This way it's easy to parse since it's just a param that you can read, and it's safe because we explicitly set this up in the scope permissions

Reply:
Nice, I like that suggestion, I'll give it a spin.